CMakeLists.txt

```
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.16.0)
+
+if(APPLE)
+	set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9" CACHE STRING "Minimum macOS deployment version")
+endif()
+
+# handle debug lib naming based on shipped project naming
+if(NOT CMAKE_DEBUG_POSTFIX)
+    set(CMAKE_DEBUG_POSTFIX d)
+endif()
+
+project(mcpp VERSION 2.7.2.14)
+
+add_library(mcpp STATIC
+	"config.h"
+	"configed.H"
+	"directive.c"
+	"eval.c"
+	"expand.c"
+	"internal.H"
+	"main.c"
+	"mbchar.c"
+	"support.c"
+	"system.c"
+	"system.H"
+)
+
+target_sources(mcpp PUBLIC
+	"${CMAKE_CURRENT_SOURCE_DIR}/mcpp_lib.h"
+	"${CMAKE_CURRENT_SOURCE_DIR}/mcpp_out.h"
+)
+
+if(UNIX)
+    include(GNUInstallDirs)
+    
+    set_property(
+        TARGET mcpp
+        PROPERTY PUBLIC_HEADER "mcpp_lib.h" "mcpp_out.h"
+    )
+
+	target_compile_options(mcpp PRIVATE "-O2")
+
+	if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+		set_property(
```

Should `TARGET mcpp` be in line with `set_property` to be consistent.

```suggestion
              set_property(TARGET mcpp
```